include/mitsuba/render/shape.h

```diff
@@ -679,6 +679,45 @@ class MI_EXPORT_LIB Shape : public Object {
      */
     virtual Float surface_area() const;
 
+    /**
+     * \brief Add a texture attribute with the given \c name.
+     *
+     * If an attribute with the same name already exists, it is replaced.
+     *
+     * Note that \c Mesh shapes can additionally handle per-vertex
+     * and per-face attributes via the \c Mesh::add_attribute method.
+     *
+     * \param name
+     *     Name of the attribute
+     * \param texture
+     *     Texture to store. The dimensionality of the attribute
+     *     is simply the channel count of the texture.
+     */
+    virtual void add_texture_attribute(const std::string &name, Texture *texture);
+
+    /// Return the texture attribute associated with \c name.
+    Texture *texture_attribute(const std::string &name) {
+        auto it = m_texture_attributes.find(name);
```

Comment:
I think that this kind of more costly code (hash table or map lookup) should go into the C++ file, there is no performance benefit from inlining it to the caller. (It's possible that this rule is inconsistenly followed).

Reply:
Good point, moving both `Shape::texture_attribute()` and `Mesh::attribute_buffer()` implementations to their respective `.cpp`.